pkg/util/ssh.go

```diff
@@ -174,6 +174,13 @@ func runSSHCommand(dialer sshDialer, cmd, user, host string, signer ssh.Signer)
 		Auth: []ssh.AuthMethod{ssh.PublicKeys(signer)},
 	}
 	client, err := dialer.Dial("tcp", host, config)
+	// Retry the dial a few times if it failed.
+	var retries = 3
```

Comment:
wait.Poll doesn't depend on the ssh functions; just the util package (this is one of the problems with util packages btw). You could break the ssh functions out of the util package and into a separate package by itself to break the cycle.